lib/OptimizationNLPModels/src/OptimizationNLPModels.jl

```diff
@@ -0,0 +1,42 @@
+module OptimizationNLPModels
+
+using Reexport
+@reexport using NLPModels, Optimization, ADTypes
+
+function SciMLBase.OptimizationFunction(nlpmodel::AbstractNLPModel,
+        adtype::ADTypes.AbstractADType = SciMLBase.NoAD(); kwargs...)
+    f(x, p) = NLPModels.obj(nlpmodel, x)
+    grad(G, u, p) = NLPModels.grad!(nlpmodel, u, G)
+    hess(H, u, p) = (H .= NLPModels.hess(nlpmodel, u))
+    hv(Hv, u, v, p) = NLPModels.hprod!(nlpmodel, u, v, Hv)
+
+    if !unconstrained(nlpmodel)
+        cons(res, x, p) = NLPModels.cons!(nlpmodel, x, res)
+        cons_j(J, x, p) = (J .= NLPModels.jac(nlpmodel, x))
+        cons_jvp(Jv, v, x, p) = NLPModels.jprod!(nlpmodel, x, v, Jv)
+
+        return OptimizationFunction(
+            f, adtype; grad, hess, hv, cons, cons_j, cons_jvp, kwargs...)
+    end
+
+    return OptimizationFunction(f, adtype; grad, hess, hv, kwargs...)
+end
+
+function OptimizationProblem(nlpmodel::AbstractNLPModel,
+        adtype::ADTypes.AbstractADType = SciMLBase.NoAD(); kwargs...)
+    f = OptimizationFunction(nlpmodel, adtype; kwargs...)
+    u0 = nlpmodel.meta.x0
+    lb = nlpmodel.meta.lvar
+    ub = nlpmodel.meta.uvar
```

Comment:
In cases where these bounds are not specified, what's the default? Since we'd want `nothing` in those cases we should confirm whether or not this can be used directly (same for the constraints' bounds)